src/components/transaction/GetTransactionsId.tsx

```diff
@@ -23,45 +30,39 @@ const GetTransactionsId = () => {
           value={accountId}
           placeholder="ccc3a91d-449c-41ff-a6fe-d79001431e4f"
           onChange={(e) => setAccountId(e.target.value)}
-        ></input>
+        />
         <button
           onClick={() => {
             const validateInput = validateAccountId(accountId)
             if (validateInput) {
               setErrorMessage(validateInput)
-              setRequestStatus('error')
+
+              //to prevent fetching/refetching with invalid uuid
+              setSubmittedAccountId('')
             } else {
-              setRequestStatus('fetching')
-              api
-                .get(`accounts/${accountId}/transactions`)
-                .json<Array<TransactionResponse>>()
-                .then((data) => {
-                  setTransactionsForId(data)
-                  setRequestStatus('success')
-                })
-                .catch((e) => {
-                  setRequestStatus('error')
-                  if (e instanceof HTTPError) {
-                    e.response.json().then((e) => setErrorMessage(e.message))
-                  }
-                  setErrorMessage(e.message)
-                })
+              setSubmittedAccountId(accountId)
+              //must refetch since transaction won't show up if it is searched for,
+              //created, then searched for again; stays stuck on
+              //"No transactions found" unless manually refetched
+              getTransactionsForId.refetch()
+              setErrorMessage(null)
             }
           }}
         >
           Search
         </button>
 
         <div className="requestStatus">
-          {requestStatus === 'fetching' || requestStatus === 'error' ? (
-            <FormStatus
-              request={{ status: requestStatus, message: errorMessage }}
-            />
+          {getTransactionsForId.isLoading ? (
+            <div className="loading">Please wait...</div>
+          ) : errorMessage != null ? (
```

Comment:
remember we always want strict equals
```suggestion
          ) : errorMessage !== null ? (
```